src/sentry/api/base.py

```diff
@@ -205,6 +206,23 @@ def build_cursor_link(self, request: Request, name: str, cursor: Cursor):
     def convert_args(self, request: Request, *args, **kwargs):
         return (args, kwargs)
 
+    def permission_denied(self, request, message=None, code=None):
+        """
+        Raise a specific superuser exception if the user can become superuser
+        and the only permission class is SuperuserPermission. Otherwise, raises
+        the appropriate exception according to parent DRF function.
+        """
+        permissions = self.get_permissions()
+
+        can_be_superuser = request.user.is_authenticated and request.user.is_superuser
+        sole_superuser_permission = len(permissions) == 1 and isinstance(
```

Comment:
```suggestion
        has_only_superuser_permission = len(permissions) == 1 and isinstance(
```
we're doing this because at the moment we cannot do an OR with permission classes, right? will we come back to this if/when people start using the OR (after the drf upgrade)?

Reply:
We're doing this b/c we only want to raise `SuperuserRequired` if it's the only permission class. Otherwise, you could have composed permissions so indicating that you NEED superuser would be wrong.

I think after the OR, we shouldn't actually need to update this b/c the check will fail which is what we want

Reply:
i'm just wondering is this a problem for permissions like this? https://github.com/getsentry/sentry/blob/69a319366a26fe1a127196b3010aec8985af0420/src/sentry/api/endpoints/project_performance_issue_settings.py#L102-L106

in which we check superuser OR some other permission in the permission class already

would we raise `SuperuserRequired` in this case?

Reply:
No we shouldn't in this case because you can pass the permission check using the normal project permissions. Also I need to talk to the person who wrote that permission because I think the superuser permission part is redundant